### PR TITLE
Test and Documents

### PR DESCRIPTION
**Pull request by astj**

### CODE REVIEW

**Comment by astj** [comment]
I Should mention `unstub`

**Comment by aereal** [comment]
This patch may fixes the problem about `unstub`:

```diff
diff --git a/lib/Test/WWW/Stub.pm b/lib/Test/WWW/Stub.pm
index c47000a..7c01a26 100644
--- a/lib/Test/WWW/Stub.pm
+++ b/lib/Test/WWW/Stub.pm
@@ -94,11 +94,11 @@ my $app = sub {
     return [ 499, [], [] ];
 };
 
-LWP::Protocol::PSGI->register($app);
+my $regiser_g = LWP::Protocol::PSGI->register($app);
 
 sub unstub {
     Carp::croak 'guard is required' unless defined wantarray;
-    LWP::Protocol::PSGI->unregister;
+    undef $regiser_g;
     return guard {
         LWP::Protocol::PSGI->register($app);
     }
```

**Comment by astj** [comment]
@aereal Thanks! Implemented as https://github.com/astj/p5-Test-WWW-Stub/commit/ba569dc2a23fb9d2630997e7bf6f184ac58c9888 .

**Comment by aereal** [comment]
:hotsprings:

File: lib/Test/WWW/Stub.pm [hunk 1]
```diff
@@ -111,15 +111,107 @@ __END__
...
     use Test::WWW::Stub;
 
+    my $ua = LWP::UserAgent->new;
+
+    my $stubbed_res = [ 200, [], ['okay'] ];
+
+    {
+        my $guard = Test::WWW::Stub->register(q<http://example.com/TEST/>, $stubbed_res);
+
+        is $ua->get('http://example.com/TEST')->content, 'okay';
```
Comment (hitode909) [inline lib/Test/WWW/Stub.pm]: L126 stubs `/TEST/`, but L128 access `/TEST`.

Comment (astj) [inline lib/Test/WWW/Stub.pm]: Thanx! fixed on https://github.com/astj/p5-Test-WWW-Stub/commit/3b60bb513945d66835cfe91e8f0b3f1b8bdf910f

**Comment by astj** [comment]
Seems okay

**Pull request merged**